python/plugins/processing/algs/gdal/ClipRasterByMask.py

```diff
@@ -81,9 +94,26 @@ def initAlgorithm(self, config=None):
                                                         self.tr('Match the extent of the clipped raster to the extent of the mask layer'),
                                                         defaultValue=True))
         self.addParameter(QgsProcessingParameterBoolean(self.KEEP_RESOLUTION,
-                                                        self.tr('Keep resolution of output raster'),
+                                                        self.tr('Keep resolution of input raster'),
                                                         defaultValue=False))
-
+        self.addParameter(QgsProcessingParameterBoolean(self.SET_RESOLUTION,
```

Comment:
To me this parameter seems redundant -- we could cover the same functionality by testing if X_RESOLUTION and Y_RESOLUTION are set and not None. (and throw an exception if only one is set and not the other)